tools/importer-rest-api-specs/components/dataapigeneratorjson/models/operations.go

```diff
@@ -0,0 +1,73 @@
+package models
+
+type Operation struct {
+	// Name specifies the Display Name for this Operation (e.g. `Create`) which is also
+	// valid as an identifier.
+	Name string `json:"Name"`
+
+	// ContentType specifies the format of the information being sent with the Operation (e.g. `application/json; charset=utf-8`)
+	ContentType string `json:"ContentType,omitempty"`
+
+	// ExpectedStatusCodes specifies is a list of Status Codes which are expected to be returned (e.g. 200, 201)
+	ExpectedStatusCodes []int `json:"ExpectedStatusCodes,omitempty"`
+
+	// FieldContainingPaginationDetails is a reference to the field within the Response
+	// which contains the pagination details, (e.g. `nextLink`)
+	FieldContainingPaginationDetails string `json:"FieldContainingPaginationDetails,omitempty"`
+
+	// LongRunning specifies if this is a Long Running Operation, meaning that Clients
+	// should follow any `Location` headers to track the result of this operation
+	LongRunning bool `json:"LongRunning,omitempty"`
+
+	// HTTPMethod is the Method used for this operation, (e.g. `GET`, `POST`)
+	HTTPMethod string `json:"HTTPMethod,omitempty"`
+
+	// todo do we need this OptionsObject as well as Options?
+	OptionsObject OptionsObject `json:"OptionsObject,omitempty"`
+
+	// Options is a list of options which can be specified for this operation
+	// these are querystring parameters such as 'limit' or 'forceDelete' or similar
+	Options []Option `json:"Options,omitempty"`
+
+	// ResourceId specifies the optional Resource ID used for this operation
+	ResourceId string `json:"ResourceId,omitempty"`
+
+	// RequestObject specifies the optional ObjectDefinition to be specified in the Request
+	RequestObject ObjectDefinition `json:"RequestObject,omitempty"`
+
+	// RequestObject specifies the optional ObjectDefinition to be returned by the Request
+	ResponseObject ObjectDefinition `json:"ResponseObject,omitempty"`
+
+	// UriSuffix specifies the suffix which should be appended to the ResourceID for this operation
+	// (e.g. `/shutdown`)
+	UriSuffix string `json:"UriSuffix,omitempty"`
```

Comment:
all 4 of these can be optional - so should be pointers:

```suggestion
	ResourceId *string `json:"ResourceId,omitempty"`

	// RequestObject specifies the optional ObjectDefinition to be specified in the Request
	RequestObject *ObjectDefinition `json:"RequestObject,omitempty"`

	// RequestObject specifies the optional ObjectDefinition to be returned by the Request
	ResponseObject *ObjectDefinition `json:"ResponseObject,omitempty"`

	// UriSuffix specifies the suffix which should be appended to the ResourceID for this operation
	// (e.g. `/shutdown`)
	UriSuffix *string `json:"UriSuffix,omitempty"`
```

`ResourceId` likely should become `ResourceIdName` so that it's clearer this is the name and not the URI etc?